### DISCUSSION TITLE
unable to compile jack2 for windows, on xubuntu. need help

### DISCUSSION THREAD

**Discussion by DCTech-lab** [jackaudio/jack2#904] [Q&A] [answered]
Hi, 

I am new to jack and was trying to compile jack2 for windows, on xubuntu. I followed the steps in build.yml (win32 section). it goes smoothly till linking the jackserver.dll, where it throws error "lto1: error: unknown pass auto-import specified in ‘-fdisable’" and compilation stops. 

giving next step for msvc dll buid it says that .def files for the said dlls are missing. I couldnt find the files in repo. so looks like these are generated on fly. But as previous step failed, its not generated.

can someone help me to rectify the issue in compilation?

Thanks
Durgesh Kumar

**Comment by falkTX** [comment, marked as answer]
JACK is not compatible with LTO, that needs to be disabled

**Reply by DCTech-lab** [reply]
how to do that? on bash console command or changing some flag in linker command?

Regards
Durgesh

**Reply by DCTech-lab** [reply]
OK, its because I forgot to set environment variables before building jack. However I compiled it on windows11 using MSys2 mingw environment using help in https://github.com/DC-codes/BuildHelp/blob/main/jack2/buildjack.sh.

Thanks,
Durgesh Kumar
